pkg/event/config.go

```diff
@@ -272,15 +272,14 @@ func (conf *Config) ToRulesMap() RulesMap {
 // ParseConfig - parses data in reader to notification configuration.
 func ParseConfig(reader io.Reader, region string, targetList *TargetList) (*Config, error) {
 	var config Config
-	if err := xml.NewDecoder(reader).Decode(&config); err != nil {
-		return nil, err
-	}
+	var err error
 
-	if err := config.Validate(region, targetList); err != nil {
+	if err = xml.NewDecoder(reader).Decode(&config); err != nil {
 		return nil, err
 	}
 
-	config.SetRegion(region)
+	err = config.Validate(region, targetList)
 
-	return &config, nil
+	config.SetRegion(region)
+	return &config, err
```

Comment:
What does it mean to return non-nil config and error?

Reply:
Now that we should not send an nil config for an `ARN Not Found` error. the config will not be empty/nil any more. By this way we would always save the config even if one of the ARN is not found